src/main/java/org/generationcp/middleware/dao/dms/StockDao.java

```diff
@@ -439,4 +517,88 @@ private void addFixedVariableIfPresent(final TermId termId, final String value,
 				measurementVariable.get().getTermId(), new StudyEntryPropertyData(value));
 		}
 	}
+
+	private void appendVariableIdAndOperationToFilterQuery(final StringBuilder sql, final List<String> havingConditions,
+		final StudyEntrySearchDto.Filter filter,
+		final Set<String> variableIds, final boolean performLikeOperation) {
+		final Integer variableId = filter.getVariableId();
+		for (final String observableId : variableIds) {
+			if (variableId != null && !variableId.equals(Integer.valueOf(observableId))) {
+				continue;
+			}
+			if (factorsFilterHavingCollection.contains(observableId)) {
+				this.addHavingCondition(havingConditions, observableId, performLikeOperation);
+			} else {
+				final String variableTypeString = filter.getVariableTypeMap().get(observableId);
+				this.applyFactorsFilter(sql, observableId, variableTypeString, performLikeOperation);
+			}
+		}
+	}
+
+	private void applyFactorsFilter(final StringBuilder sql, final String variableId, final String variableType,
+		final boolean performLikeOperation) {
+		// Check if the variable to be filtered is in one of the columns in stock, nd_experiment, geolocation or sum of samples
+		final String observationUnitClause = VariableType.OBSERVATION_UNIT.name().equals(variableType) ? "nde.observation_unit_no" : null;
```

Comment:
seems to me that this block is not needed. we do not have observation unit, neither sample in this case @lik-83

Reply:
Done!